src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -497,8 +499,14 @@ private static TypeSymbol MakeMergedTupleType(ArrayBuilder<DeconstructionVariabl
                 return null;
             }
 
-            return TupleTypeSymbol.Create(locationOpt: null, elementTypes: typesBuilder.ToImmutableAndFree(), elementLocations: default(ImmutableArray<Location>),
-                                    elementNames: default(ImmutableArray<string>), compilation: compilation, diagnostics: diagnostics);
+            return TupleTypeSymbol.Create(
+                locationOpt: null,
+                elementTypes: typesBuilder.ToImmutableAndFree(),
+                elementLocations: default(ImmutableArray<Location>),
+                elementNames: default(ImmutableArray<string>),
+                compilation: compilation,
+                diagnostics: diagnostics,
+                shouldCheckConstraints: false);
```

Comment:
As @jcouv said:

>The tuple created by MakeMergedTupleType is not identical to that one (it represents a smaller tree of types, it is used for figuring out natural types in tuple literal). I think it is better not to check constraints here (it would report errors that are already reported later).

Constraints will be checked for the final tuple type created in `MakeDeconstructionConstructionStep`.